hail/src/main/scala/is/hail/cxx/Compile.scala

```diff
@@ -172,13 +169,11 @@ object Compile {
     mod.close()
     st.close()
 
-    val hadoopConf = new SerializableHadoopConfiguration(HailContext.get.sc.hadoopConfiguration)
+    val fs = new HadoopFS(HailContext.get.sc.hadoopConfiguration)
 
     { (region: Long, v2: Long) =>
       val st2 = new NativeStatus()
-      val jObjectArgs = new ObjectArray(sparkUtils,
-        new RichHadoopConfiguration(hadoopConf.value).asInstanceOf[AnyRef],
-        new ByteArrayInputStream(literals)).get()
+      val jObjectArgs = new ObjectArray(sparkUtils,fs,new ByteArrayInputStream(literals)).get()
```

Comment:
Not sure if fs needs to be asInstanceOf[AnyRef].

Reply:
`asInstanceOf[AnyRef]` is necessary for classes that extend `AnyVal`, since `AnyVal`s are not backed by Java `Object`s at runtime

Reply:
Ah, good to know, thanks.